nexus/src/db/datastore/disk.rs

```diff
@@ -64,19 +67,29 @@ impl DataStore {
 
         let gen = disk.runtime().gen;
         let name = disk.name().clone();
-        let disk: Disk = diesel::insert_into(dsl::disk)
-            .values(disk)
-            .on_conflict(dsl::id)
-            .do_nothing()
-            .returning(Disk::as_returning())
-            .get_result_async(self.pool())
-            .await
-            .map_err(|e| {
+        let project_id = disk.project_id;
```

Comment:
Really how neat it is that this was so straightforward (at least this part), the way `collection_insert` was built.